broken build: stm32h747xi_giga — multiple STM32 cores detected (Mbed + CMSIS-imported STM32_MCU_SERIES)

## Summary

The `stm32h747xi_giga` workflow (Arduino GIGA R1, `build_giga_r1.yml`) is currently failing on `master`. Confirmed both **locally** with `bash compile stm32h747xi` and on the **CI run** at https://github.com/FastLED/FastLED/actions/runs/27003825571 (latest commit on master at time of filing).

This is the next group from the README badge sweep started in #2779 / #2780 (which closed the four AVR-family failures).

## Failure

The error fires from `src/platforms/arm/stm32/core_detection.h:138`:

```
src/platforms/arm/stm32/core_detection.h:138:4: error:
  #error "Multiple STM32 Arduino cores detected - conflicting definitions.
          Check your build environment."
```

It also emits a `F_CPU` redefinition warning from `src/platforms/arm/giga/led_sysdef_arm_giga.h:39` (`#define F_CPU 480000000` redefining the value the toolchain already provided on the command line). The redefinition is a warning today but should be cleaned up in the same change.

## Root cause

`core_detection.h` defines five per-core flags (`_FL_STM32_HAS_STMDUINO`, `_FL_STM32_HAS_ZEPHYR`, `_FL_STM32_HAS_MBED`, `_FL_STM32_HAS_LIBMAPLE`, `_FL_STM32_HAS_PARTICLE`) and validates `_FL_STM32_CORE_COUNT <= 1`. The legacy LibMaple flag fires on:

```cpp
#if defined(__STM32F1__) || defined(__STM32F4__) || defined(STM32_MCU_SERIES)
```

On the GIGA's Mbed-STM32H7 toolchain, **both** `ARDUINO_ARCH_MBED` _and_ `STM32_MCU_SERIES` are defined — the latter is pulled in by Mbed's CMSIS/HAL layer for hardware register configuration, not because LibMaple is in use. The count goes to 2 and the hard error fires.

The elif chain in the same file already resolves the ambiguity correctly (Mbed wins over LibMaple by priority order), but the count macros don't follow that priority — so the validation rejects builds the actual core selection accepts.

## Fix

Gate the secondary-core flags (LibMaple, Particle) on absence of the modern primary cores (STM32duino, Mbed, Zephyr), matching the elif priority. Genuine misconfigurations (e.g. `ARDUINO_ARCH_STM32` + `ARDUINO_ARCH_MBED` defined simultaneously) still get caught.

Same PR cleans up the `F_CPU` redefinition by guarding with `#ifndef`.

## Acceptance criteria

- [ ] `bash compile stm32h747xi` passes on master
- [ ] CI `build_giga_r1.yml` goes green on master
- [ ] No regressions on the other STM32-family compiles in `bash compile --supported-boards` (`stm32f103c8`, `stm32f103cb`, `stm32f103tb`, `stm32f411ce`, `nucleo_f429zi`, `nucleo_f439zi`)
- [ ] The `F_CPU` redefinition warning is gone

## Out of scope

- The 7 remaining red badges from #2779 (apollo3 ×2, rp2040, two size checks, top-level `build`, `uno AVR8JS Test`). Those need their own root-cause work and will be filed separately.

---

_Next badge group after #2780 (AVR watchdog). Filed via `/loop` README badge sweep._

--- a/src/platforms/arm/giga/led_sysdef_arm_giga.h
+++ b/src/platforms/arm/giga/led_sysdef_arm_giga.h
@@ -36,6 +36,10 @@
 typedef volatile fl::u32 RoReg;
 typedef volatile fl::u32 RwReg;
 
+// Mbed-on-STM32H7 toolchains often supply `F_CPU` on the command line; only
+// define it here when the build environment hasn't already.
+#ifndef F_CPU
 #define F_CPU 480000000
+#endif
 
 #endif

--- a/src/platforms/arm/stm32/core_detection.h
+++ b/src/platforms/arm/stm32/core_detection.h
@@ -27,31 +27,54 @@
 // Core Detection - Mutually Exclusive
 // ============================================================================
 
-#if defined(ARDUINO_ARCH_STM32)
-  #define _FL_STM32_HAS_STMDUINO 1
-#else
-  #define _FL_STM32_HAS_STMDUINO 0
-#endif
+// Each per-core flag mirrors the priority of the elif chain below — Zephyr
+// wins over Mbed wins over STM32duino wins over LibMaple wins over Particle.
+// This priority gating matters because Arduino frameworks define each other's
+// compat identifiers: Mbed-on-STM32 (e.g. Arduino GIGA R1) defines
+// `ARDUINO_ARCH_STM32`, and its CMSIS layer pulls in `STM32_MCU_SERIES`.
+// Without the priority gate the validation below would false-positive on
+// legitimate single-core builds. With the gate, genuine misconfigurations
+// (e.g. two frameworks' libraries linked into the same image with neither
+// taking precedence) still produce a count > 1 and fire the hard error.
 
+// Zephyr — highest priority. Requires ARDUINO_ARCH_ZEPHYR + UNO Q identifier.
 #if defined(ARDUINO_ARCH_ZEPHYR) && (defined(ARDUINO_UNO_Q) || defined(CONFIG_BOARD_ARDUINO_UNO_Q) || defined(CONFIG_SOC_STM32U585XX))
   #define _FL_STM32_HAS_ZEPHYR 1
 #else
   #define _FL_STM32_HAS_ZEPHYR 0
 #endif
 
-#if defined(ARDUINO_ARCH_MBED)
+// Mbed — second priority. Counted whenever ARDUINO_ARCH_MBED is set and
+// the higher-priority Zephyr selection isn't already taken.
+#if defined(ARDUINO_ARCH_MBED) && !_FL_STM32_HAS_ZEPHYR
   #define _FL_STM32_HAS_MBED 1
 #else
   #define _FL_STM32_HAS_MBED 0
 #endif
 
-#if defined(__STM32F1__) || defined(__STM32F4__) || defined(STM32_MCU_SERIES)
+// STM32duino — third priority. Only counted when no higher-priority modern
+// framework (Mbed, Zephyr) is also detected. Mbed-on-STM32 also defines
+// ARDUINO_ARCH_STM32 as a compat macro, so the bare presence of that
+// identifier doesn't on its own mean STM32duino is in use.
+#if defined(ARDUINO_ARCH_STM32) && !_FL_STM32_HAS_MBED && !_FL_STM32_HAS_ZEPHYR
+  #define _FL_STM32_HAS_STMDUINO 1
+#else
+  #define _FL_STM32_HAS_STMDUINO 0
+#endif
+
+// LibMaple and Particle — secondary cores. Their identifier macros
+// (`__STM32F1__`, `__STM32F4__`, `STM32_MCU_SERIES`, `PLATFORM_ID`, `SPARK`)
+// can also leak in from modern frameworks' CMSIS/HAL layers, so they only
+// count when no modern framework is detected.
+#if (defined(__STM32F1__) || defined(__STM32F4__) || defined(STM32_MCU_SERIES)) \
+    && !_FL_STM32_HAS_STMDUINO && !_FL_STM32_HAS_MBED && !_FL_STM32_HAS_ZEPHYR
   #define _FL_STM32_HAS_LIBMAPLE 1
 #else
   #define _FL_STM32_HAS_LIBMAPLE 0
 #endif
 
-#if defined(PLATFORM_ID) || defined(SPARK)
+#if (defined(PLATFORM_ID) || defined(SPARK)) \
+    && !_FL_STM32_HAS_STMDUINO && !_FL_STM32_HAS_MBED && !_FL_STM32_HAS_ZEPHYR
   #define _FL_STM32_HAS_PARTICLE 1
 #else
   #define _FL_STM32_HAS_PARTICLE 0

